### ISSUE TITLE
[spi_device] Flash Mode and CMD_INFO

### ISSUE THREAD

**Issue by kosta-kojdic** [lowRISC/opentitan#11491]
We have read status command in flash mode without any cmd_info configured. On passthrough_o output we have data and csb assertion, it propagates data downstream. Is it expected behavior? If not, can you summarize expected flow and behavior for flash commands? To reproduce from this PR #11220  :
./util/dvsim/dvsim.py hw/ip/spi_device/dv/spi_device_sim_cfg.hjson -i spi_device_read_status --fixed-seed 6 --tool xcelium --waves --xprop-off -v h --max-parallel 1

**Comment by eunchan** [comment]
that is expected if `passthrough_o.passthrough_en` is 0. Intention is to reduce the timing as much as possible and SPI_HOST to put muxes.
